resources/usr/bin/openhab

```diff
@@ -0,0 +1,158 @@
+  echo "    start                       -- Starts openHAB as a daemon service."
+  echo "    stop                        -- Stops an openHAB daemon service."
+  echo "    restart                     -- Stops then starts an openHAB daemon service."
+  echo "    status                      -- Checks to see if openHAB is running."
+  echo "    console                     -- Opens the openHAB console."
+  echo "    backup [filename]           -- Stores the current configuration of openHAB."
+  echo "    restore filename            -- Restores the state of openHAB to a backup."
+  echo "    showlogs                    -- Displays the log messages of openHAB as they appear."
+  echo ""
+  exit 2
+}
+
+# Script requires options, list them if none given
+if [ "$#" -eq "0" ]; then
+  usage
+fi
+
+# Select between common daemon services
+# Scripts are included with package
+OHStart() {
+  if [ -x /bin/systemctl ] ; then
+    echo "Using command '/bin/systemctl start openhab2.service'..."
+    /bin/systemctl start openhab2.service
+  elif [ -x "/etc/init.d/openhab2" ]; then
+    invokeCommand="$(which invoke-rc.d 2>/dev/null)"
+    if [ -x "$invokeCommand" ]; then
+      echo "Using command 'invoke-rc.d openhab2 start'..."
+      invoke-rc.d openhab2 start
+    else
+      /etc/init.d/openhab2 start
+    fi
+  fi
+}
+
+# Select between common services to stop openHAB
+OHStop() {
+  if [ -x /bin/systemctl ]; then
+    echo "Using command '/bin/systemctl --no-reload stop openhab2.service'..."
+    /bin/systemctl --no-reload stop openhab2.service
+  elif [ -x "/etc/init.d/openhab2" ]; then
+    invokeCommand="$(which invoke-rc.d 2>/dev/null)"
+    if [ -x "$invokeCommand" ]; then
+      echo "Using command 'invoke-rc.d openhab2 stop'..."
+      invoke-rc.d openhab2 stop
+    else
+      /etc/init.d/openhab2 stop
+    fi
+    # init.d script needs time to fully stop the service
+    sleep 5
+  fi
+}
+
+# Some functions require root
+checkRoot() {
+  if [ "$(id -u)" -ne 0 ]; then
+    optionName="$1"
+    echo "This option needs to run as root! (e.g. use sudo openhab $optionName ..)" >&2
+    exit 1
+  fi
+}
+
+# Load any possible custom directory
+if [ -r /etc/default/openhab2 ]; then
+  . /etc/default/openhab2
+fi
+
+# Read and set openHAB variables set in /etc/default/openhab
+# Otherwise, use the defaults set from package install
+if [ -z "${OPENHAB_HOME}" ];       then OPENHAB_HOME="/usr/share/openhab2"; fi
+if [ -z "${OPENHAB_RUNTIME}" ];    then OPENHAB_RUNTIME="${OPENHAB_HOME}/runtime"; fi
+if [ -z "${OPENHAB_LOGDIR}" ];     then OPENHAB_LOGDIR="/var/log/openhab2"; fi
+export OPENHAB_HOME
+export OPENHAB_RUNTIME
+export OPENHAB_LOGDIR
+
+option="$1"
+shift
+
+# Select between options
+case "$option" in
+  "start")
+    checkRoot "$option"
+    OHStart
+    ;;
+  "stop")
+    checkRoot "$option"
+    OHStop
+    ;;
+  "restart")
+  checkRoot "$option"
+  if [ -x /bin/systemctl ]; then
+    echo "Using command '/bin/systemctl restart openhab2.service'..."
```

Comment:
This is an excellent idea! I was worried to add another "onion layer" but this will reveal the magic clearly.

How about "Executing" instead of "Using"?